test/tools/libtesteth/boostTest.cpp

```diff
@@ -36,15 +36,17 @@ static std::ostringstream strCout;
 std::streambuf* oldCoutStreamBuf;
 std::streambuf* oldCerrStreamBuf;
 
-void createRandomTestWrapper()
+void customTestSuiteWrapper()
 {
-	//restore output for creating test
+	//restore output
 	std::cout.rdbuf(oldCoutStreamBuf);
 	std::cerr.rdbuf(oldCerrStreamBuf);
 
-	if (!dev::test::createRandomTest())
-		throw framework::internal_error("Create random test error! See std::error for more details.");
-
+	if (dev::test::Options::get().createRandomTest.is_initialized())
+	{
+		if (!dev::test::createRandomTest())
+			throw framework::internal_error("Create random test error! See std::error for more details.");
+	}
```

Comment:
Why is it ok when `createRandomTest` is not initialized?

Reply:
I see. I was hoping to put --singletest into this function. but then I realized that it is not clear how to link -t argument with the test suite